platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -0,0 +1,60 @@
+#import "NSExpression+MGLAdditions.h"
+
+class FilterExpressionEvaluator {
+public:
+    id operator()(mbgl::Value value) {
```

Comment:
Instead of a single `mbgl::Value` operator with a number of `if…is` clauses, there should be a separate operator for each type you want to convert. So, for example:

```objc
id operator()(bool value) {
    return @(value.get<bool>());
}
```